controllers/nonadmin_controller.go

```diff
@@ -86,6 +85,21 @@ func (r *DPAReconciler) ReconcileNonAdminController(log logr.Logger) (bool, erro
 		return true, nil
 	}
 
+	selector, err := fields.ParseSelector(fmt.Sprintf("metadata.namespace!=%s", r.NamespacedName.Namespace))
+	if err != nil {
+		return false, err
+	}
+	dpaList := &oadpv1alpha1.DataProtectionApplicationList{}
+	err = r.List(r.Context, dpaList, &client.ListOptions{FieldSelector: selector})
```

Comment:
A corner case here should be if 2 (or more) DPAs are created at the same time. But since DPA reconciles every minute, that's not to bad? (DPAs would have error message in their statuses, but multiple NAC would also exist in the cluster. To really avoid this, before erroring out, could delete NAC deployment)

Another question here: since DPA reconciles every minute, its too much to do a cluster wide get call every minute?

Reply:
Or a more drastic change would be to move this controller out of the current ReconcileBatch into its own thing.

Reply:
thinking again, I think we do not need to worry about having multiple NAC deployments after validation 

- If 2 DPAs with NAC enabled are created at the same time, when they reach this part, both of them will not deploy NAC

- If one DPA is created and after some time another is created, only first DPA NAC will be deployed

Right?